symengine/eval_double.cpp

```diff
@@ -240,6 +240,10 @@ class EvalDoubleVisitor : public BaseVisitor<C>
         } else if (eq(x, *EulerGamma)) {
             result_ = 0.5772156649015328606065; // use until polygamma or
                                                 // digamma is implemented
+        } else if (eq(x, *Catalan)) {
+            result_ = 0.9159655941772190150546;
+        } else if (eq(x, *GoldenRatio)) {
+            result_ = 1.6180339887498948482045;
```

Comment:
but x is an object of derived class(Constant ) and eval_double takes object of base class(basic) ..